src/lib/logical_query_plan/mock_node.hpp

```diff
@@ -6,28 +6,41 @@
 #include <vector>
 
 #include "abstract_lqp_node.hpp"
+#include "all_type_variant.hpp"
 
 namespace opossum {
 
 /**
- * Node that represents a table that has no data backing it, but may provide (mocked) statistics. Intended for usage in
- * tests (e.g. general LQP tests, optimizer tests that just rely on statistics and not actual data) and the playground
+ * Node that represents a table that has no data backing it, but may provide
+ *  - (mocked) statistics
+ *  - or just a column layout. It will pretend it created the columns.
+ * It is useful in tests (e.g. general LQP tests, optimizer tests that just rely on statistics and not actual data) and
+ * the playground
  */
 class MockNode : public AbstractLQPNode {
  public:
-  MockNode();
-  explicit MockNode(const std::shared_ptr<TableStatistics>& statistics);
+  using ColumnDefinitions = std::vector<std::pair<DataType, std::string>>;
```

Comment:
I see a value here, and this does not need to change. Still, I want to comment that hiding everything in aliases does not necessarily improve readability. While it makes the code shorter, it increases the number of indirections that others need to keep track of.

Reply:
Yeah, I'd probably agree. Just hate things like `std::vector<std::shared_ptr<Something>>`...